### DISCUSSION TITLE
Create new, universal my account hook

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#33421] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

We currently have 4 places to show additional links in my account zone and every one of them has a different syntax. And bending it is not the best way to do it.

![myaccount_720](https://user-images.githubusercontent.com/6097524/144573660-dd86240e-34c8-49b5-8d7d-8b40db90d49f.png)

### Describe the solution you'd like

- We now have two hooks - displayMyAccountBlock and displayCustomerAccount.
- We need to implement a new universal hook, that would add items everywhere. This hook can be called just **displayMyAccount**.
- Modules would pass an array with link, name, icon and description, instead of the full code. Then we can play with it as we like.

### Additional context

Some small discussion went here already - https://github.com/PrestaShop/PrestaShop/issues/25024#issuecomment-865193693

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by hibatallahAouadni** [comment]
> ### Describe the solution you'd like
>
>- We now have two hooks - displayMyAccountBlock and displayCustomerAccount.
>- We need to implement a new universal hook, that would add items everywhere. This hook can be called just **displayMyAccount**.
>- Modules would pass an array with link, name, icon and description, instead of the full code. Then we can play with it as we like.

Ping @PrestaShop/prestashop-core-developers wdyt of @Hlavtox solution?

**Comment by NeOMakinG** [comment]
I agree, that would be simpler to maintain, it would be overridable by theme anyway by overriding the module 👍

**Comment by matks** [comment]
> We need to implement a new universal hook, that would add items everywhere. This hook can be called just displayMyAccount.

Yes and no.

If you provide one hook and it's used in 4 places, then when a module developer wants to display something "in places 1 and 3" and nothing on place 4 but something else on place 2, it does not work.

I think if there is 4 area, there must be one way for module developers to choose exactly what to display for each of the locations.
It can be 4 different hooks, or it can be 1 hook which can be provided a clear set of parameters that allow to control what happens.

I think 4 hooks are easier. 1 place, 1 hook. It's simple.

**Comment by NeOMakinG** [comment]
> > We need to implement a new universal hook, that would add items everywhere. This hook can be called just displayMyAccount.
> 
> Yes and no.
> 
> If you provide one hook and it's used in 4 places, then when a module developer wants to display something "in places 1 and 3" and nothing on place 4 but something else on place 2, it does not work.
> 
> I think if there is 4 area, there must be one way for module developers to choose exactly what to display for each of the locations. It can be 4 different hooks, or it can be 1 hook which can be provided a clear set of parameters that allow to control what happens.
> 
> I think 4 hooks are easier. 1 place, 1 hook. It's simple.

Good args, but the way it's done atm is hardening the template development because the module is sending the full code of the markup, this means that if the markup of the current theme changed, the style will be different from items coming from the hook.

We can probably choose both of the comments => having 4 hooks, but prodiving datas that can be used from the theme instead of pushing the markup directly (if possible, I don't know the technical part of this), wdyt?

**Comment by matks** [comment]
> (if possible, I don't know the technical part of this)

It's possible. We define the expected format and apply formatting.

**Comment by matks** [comment]
> (if possible, I don't know the technical part of this

The hook returns an Smarty array that contains the data we want. Rows of a list? Color? Labels? And we write the markup.

**Comment by Hlavtox** [comment]
I think that modules should return an array with:
- URL
- Name
- Description
- Material icon _(this is to be discussed)_
- Boolean if it should be displayed only to logged users

**Comment by Hlavtox** [comment]
bump - we need this for the new theme

**Comment by kpodemski** [comment]
@Hlavtox, isn't that something that'll make a new theme core-dependent?

**Comment by rodriciru** [comment]
I think that will be a great feature, and I think customers, and owners will be grateful with that.
Some customers won't ever enter on my account, so a list of options on the dropdown will be a great add.

I see a proble with 4 hooks. If a theme maker wants a 5 hook design, we need to add a new hook?

**Comment by kpodemski** [comment]
@Hlavtox I think that has been solved, right? 🤔 or am I mixing it with something else?

**Comment by Hlavtox** [comment]
cc @tblivet @ga-devfront
